crypto/src/lib.rs

```diff
@@ -170,6 +170,20 @@ pub trait Crypto {
         info: &[u8],
     ) -> Result<Self::Cdi, CryptoError>;
 
+    /// Derive a CDI for an exported private key based on the current base CDI and measurements
+    ///
+    /// # Arguments
+    ///
+    /// * `algs` - Which length of algorithms to use.
+    /// * `measurement` - A digest of the measurements which should be used for CDI derivation
+    /// * `info` - Caller-supplied info string to use in CDI derivation
+    fn derive_exported_cdi(
+        &mut self,
+        algs: AlgLen,
+        measurement: &Digest,
+        info: &[u8],
+    ) -> Result<Self::Cdi, CryptoError>;
```

Comment:
nit: the calling code is expecting a raw buffer with some data it can return in the response. Should be return an array type instead? We can still leave it up to the platform to fill it with a random handle (v.s. an actual CDI).

Reply:
Alternatively, we could add a function that turns a Self::Cdi into a raw buffer. And leave a comment that the platform can either populate with a raw CDI or a handle to a CDI. Not sure which you would prefer.

Reply:
I didn't understand, do you mind sharing additional context?

I think it's okay as is because the `CDI` is passed to `derive_key_pair_exported`, the caller shouldn't need to know it's underlying type.

Reply:
As discussed offline, in a follow-up PR we can have a PR for extracting the CDI buffer to return.